events/migrations/0080_populate_tsvectors_and_indices.py

```diff
@@ -0,0 +1,86 @@
+
+                                "CREATE FUNCTION events_english_content_trigger_function() RETURNS trigger AS $$ "
+                                "begin "
+                                  "new.search_vector_en := "
+                                     "setweight(to_tsvector('pg_catalog.english', coalesce(new.name_en,'')), 'A') || "
+                                     "setweight(to_tsvector('pg_catalog.english', coalesce(new.short_description_en,'')), 'B') || "
+                                     "setweight(to_tsvector('pg_catalog.english', coalesce(new.description_en,'')), 'C');"
+                                  "return new; "
+                                "end "
+                                "$$ LANGUAGE plpgsql; "
+
+                                "CREATE TRIGGER events_finnish_content_trigger BEFORE INSERT OR UPDATE ON events_event FOR EACH ROW EXECUTE PROCEDURE events_finnish_content_trigger_function();"
+                                "CREATE TRIGGER events_swedish_content_trigger BEFORE INSERT OR UPDATE ON events_event FOR EACH ROW EXECUTE PROCEDURE events_swedish_content_trigger_function();"
+                                "CREATE TRIGGER events_english_content_trigger BEFORE INSERT OR UPDATE ON events_event FOR EACH ROW EXECUTE PROCEDURE events_english_content_trigger_function();"
+                               ],
+                          
+                          reverse_sql=[
+                                       "DROP TRIGGER events_finnish_content_trigger ON events_event;"
+                                       "DROP FUNCTION events_finnish_content_trigger_function;"
+                                       "DROP TRIGGER events_swedish_content_trigger ON events_event;"
+                                       "DROP FUNCTION events_swedish_content_trigger_function;"
+                                       "DROP TRIGGER events_english_content_trigger ON events_event;"
+                                       "DROP FUNCTION events_english_content_trigger_function;"
+                                       ]
+                )
+            ]
+        ),
+        migrations.SeparateDatabaseAndState(
+            database_operations=[
+                migrations.RunSQL(sql="CREATE INDEX CONCURRENTLY events_finnish_content_index ON events_event USING GIN (search_vector_fi);",
+                                  reverse_sql="DROP INDEX events_finnish_content_index;"
+                ),
+                migrations.RunSQL(sql="CREATE INDEX CONCURRENTLY events_swedish_content_index ON events_event USING GIN (search_vector_sv);",
+                                  reverse_sql="DROP INDEX events_swedish_content_index;"
+                ),
+                migrations.RunSQL(sql="CREATE INDEX CONCURRENTLY events_english_content_index ON events_event USING GIN (search_vector_en);",
+                                  reverse_sql="DROP INDEX events_english_content_index;"
+                )
+            ]
+        )
+    ]
```

Comment:
Is there some tests for this? At least it should be really carefully tested manually, but also unit test could be written. For me as an outsider this seems overwhelming and super hard to understand what happens here. So hard actually, that I won't even try :)